src/deposit.cairo

```diff
@@ -172,6 +174,25 @@ mod Deposit {
                 self.ekubo_core.read(), @swap_data
             )
         }
+        fn repay_vaults(ref self: ContractState, supply_token: ContractAddress, vaults: Span<VaultRepayData>){
+            for vault in vaults {
+                let vault_disp = IVaultDispatcher{contract_address: (*vault).vault};
+                let vault_token = vault_disp.get_vault_token();
+                if vault_token == supply_token {
+                    continue;
+                }
+                let zk_market = self.zk_market.read();
+                let mut amount = (*vault).amount;
+                if amount == 0 {
+                    zk_market.withdraw_all(vault_token);
+                    amount = ERC20ABIDispatcher { contract_address: get_contract_address() }
+                        .balanceOf(self.owner());
```

Comment:
You are supposed to create the ERC20 dispatcher for the current `vault_token` and then check the current contract's balance to know the available amount.